pkg/istio-agent/sds-agent.go

```diff
@@ -353,13 +357,20 @@ func newSecretCache(serverOptions sds.Options) (workloadSecretCache *cache.Secre
 				log.Info("Using citadel CA for SDS")
 				serverOptions.CAEndpoint = "istio-citadel.istio-system:8060"
 			} else {
-				rootCert, err = ioutil.ReadFile(k8sCAPath)
-				if err != nil {
-					log.Warnf("Failed to load K8S cert, assume IP secure network: %v", err)
-					serverOptions.CAEndpoint = "istiod.istio-system.svc:15010"
-				} else {
-					log.Info("Using default istiod CA, with K8S certificates for SDS")
+				rootCert, err = ioutil.ReadFile(CitadelCACertPath + "/" + bootstrap.CACertNamespaceConfigMapDataName)
+				if err == nil {
+					log.Info("istiod uses Citadel issued certificate")
```

Comment:
Please stop using 'citadel' in names or messages. It can be an external CA, etc - citadel is going away, it's an internal code name.

Reply:
Revised.